fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMutualTest.java

```diff
@@ -451,6 +455,132 @@ void testMutualIndexingCrashAndAllowContinueNonMutually(boolean explicit) {
         assertAllValidated(indexes);
     }
 
+    @ParameterizedTest
+    @BooleanSource
+    void testMultiTargetContinueAsMutual(boolean explicit) {
+        // Start building with multi target, stop, continue as mutual
+        // Make sure that the mutual indexing is unblocked
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexD", new GroupingKeyExpression(EmptyKeyExpression.EMPTY, 0), IndexTypes.COUNT));
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+
+        int numRecords = 32;
+        populateData(numRecords);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        // Build as mutual, crash half way
+        final String testThrowMsg = "Intentionally crash during test";
+        try (OnlineIndexer indexBuilder = newIndexerBuilder(indexes)
+                .setLimit(10)
+                .setConfigLoader(old -> {
+                    throw new RecordCoreException(testThrowMsg);
+                })
+                .build()) {
+            RecordCoreException e = assertThrows(RecordCoreException.class, indexBuilder::buildIndex);
+            assertTrue(e.getMessage().contains(testThrowMsg));
+        }
+
+        // Continue as mutual
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        try (OnlineIndexer indexBuilder = newIndexerBuilder(indexes, timer)
```

Comment:
Since the goal of this change is to avoid losing the work done by the first indexer, it seems like this test should assert that the mutual indexing has to index some records, but not all the records.

Reply:
✅